// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: ozonmp/education_person_api/v1/education_person_api.proto

package education_person_api

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
)

// Validate checks the field values on Person with the rules defined in the
// proto definition for this message. If any rules are violated, an error is returned.
func (m *Person) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for Id

	// no validation rules for FistName

	// no validation rules for MiddleName

	// no validation rules for LastName

	if v, ok := interface{}(m.GetBirthday()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return PersonValidationError{
				field:  "Birthday",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	// no validation rules for Sex

	// no validation rules for Education

	if v, ok := interface{}(m.GetCreated()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return PersonValidationError{
				field:  "Created",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	return nil
}

// PersonValidationError is the validation error returned by Person.Validate if
// the designated constraints aren't met.
type PersonValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e PersonValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e PersonValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e PersonValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e PersonValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e PersonValidationError) ErrorName() string { return "PersonValidationError" }

// Error satisfies the builtin error interface
func (e PersonValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sPerson.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = PersonValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = PersonValidationError{}

// Validate checks the field values on CreatePersonV1Request with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *CreatePersonV1Request) Validate() error {
	if m == nil {
		return nil
	}

	if v, ok := interface{}(m.GetPerson()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return CreatePersonV1RequestValidationError{
				field:  "Person",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	return nil
}

// CreatePersonV1RequestValidationError is the validation error returned by
// CreatePersonV1Request.Validate if the designated constraints aren't met.
type CreatePersonV1RequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CreatePersonV1RequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CreatePersonV1RequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CreatePersonV1RequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CreatePersonV1RequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CreatePersonV1RequestValidationError) ErrorName() string {
	return "CreatePersonV1RequestValidationError"
}

// Error satisfies the builtin error interface
func (e CreatePersonV1RequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCreatePersonV1Request.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CreatePersonV1RequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CreatePersonV1RequestValidationError{}

// Validate checks the field values on CreatePersonV1Response with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *CreatePersonV1Response) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for PersonId

	return nil
}

// CreatePersonV1ResponseValidationError is the validation error returned by
// CreatePersonV1Response.Validate if the designated constraints aren't met.
type CreatePersonV1ResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CreatePersonV1ResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CreatePersonV1ResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CreatePersonV1ResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CreatePersonV1ResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CreatePersonV1ResponseValidationError) ErrorName() string {
	return "CreatePersonV1ResponseValidationError"
}

// Error satisfies the builtin error interface
func (e CreatePersonV1ResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCreatePersonV1Response.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CreatePersonV1ResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CreatePersonV1ResponseValidationError{}

// Validate checks the field values on DescribePersonV1Request with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *DescribePersonV1Request) Validate() error {
	if m == nil {
		return nil
	}

	if m.GetPersonId() <= 0 {
		return DescribePersonV1RequestValidationError{
			field:  "PersonId",
			reason: "value must be greater than 0",
		}
	}

	return nil
}

// DescribePersonV1RequestValidationError is the validation error returned by
// DescribePersonV1Request.Validate if the designated constraints aren't met.
type DescribePersonV1RequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DescribePersonV1RequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DescribePersonV1RequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DescribePersonV1RequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DescribePersonV1RequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DescribePersonV1RequestValidationError) ErrorName() string {
	return "DescribePersonV1RequestValidationError"
}

// Error satisfies the builtin error interface
func (e DescribePersonV1RequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDescribePersonV1Request.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DescribePersonV1RequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DescribePersonV1RequestValidationError{}

// Validate checks the field values on DescribePersonV1Response with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *DescribePersonV1Response) Validate() error {
	if m == nil {
		return nil
	}

	if v, ok := interface{}(m.GetPerson()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return DescribePersonV1ResponseValidationError{
				field:  "Person",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	return nil
}

// DescribePersonV1ResponseValidationError is the validation error returned by
// DescribePersonV1Response.Validate if the designated constraints aren't met.
type DescribePersonV1ResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DescribePersonV1ResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DescribePersonV1ResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DescribePersonV1ResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DescribePersonV1ResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DescribePersonV1ResponseValidationError) ErrorName() string {
	return "DescribePersonV1ResponseValidationError"
}

// Error satisfies the builtin error interface
func (e DescribePersonV1ResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDescribePersonV1Response.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DescribePersonV1ResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DescribePersonV1ResponseValidationError{}

// Validate checks the field values on ListPersonV1Request with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListPersonV1Request) Validate() error {
	if m == nil {
		return nil
	}

	if m.GetCursor() <= 0 {
		return ListPersonV1RequestValidationError{
			field:  "Cursor",
			reason: "value must be greater than 0",
		}
	}

	return nil
}

// ListPersonV1RequestValidationError is the validation error returned by
// ListPersonV1Request.Validate if the designated constraints aren't met.
type ListPersonV1RequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListPersonV1RequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListPersonV1RequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListPersonV1RequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListPersonV1RequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListPersonV1RequestValidationError) ErrorName() string {
	return "ListPersonV1RequestValidationError"
}

// Error satisfies the builtin error interface
func (e ListPersonV1RequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListPersonV1Request.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListPersonV1RequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListPersonV1RequestValidationError{}

// Validate checks the field values on ListPersonV1Response with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListPersonV1Response) Validate() error {
	if m == nil {
		return nil
	}

	for idx, item := range m.GetPerson() {
		_, _ = idx, item

		if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return ListPersonV1ResponseValidationError{
					field:  fmt.Sprintf("Person[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	return nil
}

// ListPersonV1ResponseValidationError is the validation error returned by
// ListPersonV1Response.Validate if the designated constraints aren't met.
type ListPersonV1ResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListPersonV1ResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListPersonV1ResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListPersonV1ResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListPersonV1ResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListPersonV1ResponseValidationError) ErrorName() string {
	return "ListPersonV1ResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ListPersonV1ResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListPersonV1Response.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListPersonV1ResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListPersonV1ResponseValidationError{}

// Validate checks the field values on RemovePersonV1Request with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *RemovePersonV1Request) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for PersonId

	return nil
}

// RemovePersonV1RequestValidationError is the validation error returned by
// RemovePersonV1Request.Validate if the designated constraints aren't met.
type RemovePersonV1RequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e RemovePersonV1RequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e RemovePersonV1RequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e RemovePersonV1RequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e RemovePersonV1RequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e RemovePersonV1RequestValidationError) ErrorName() string {
	return "RemovePersonV1RequestValidationError"
}

// Error satisfies the builtin error interface
func (e RemovePersonV1RequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sRemovePersonV1Request.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = RemovePersonV1RequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = RemovePersonV1RequestValidationError{}

// Validate checks the field values on RemovePersonV1Response with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *RemovePersonV1Response) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for Ok

	return nil
}

// RemovePersonV1ResponseValidationError is the validation error returned by
// RemovePersonV1Response.Validate if the designated constraints aren't met.
type RemovePersonV1ResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e RemovePersonV1ResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e RemovePersonV1ResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e RemovePersonV1ResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e RemovePersonV1ResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e RemovePersonV1ResponseValidationError) ErrorName() string {
	return "RemovePersonV1ResponseValidationError"
}

// Error satisfies the builtin error interface
func (e RemovePersonV1ResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sRemovePersonV1Response.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = RemovePersonV1ResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = RemovePersonV1ResponseValidationError{}
